DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces_overview/mm-host/src/vector_add.cpp

```diff
@@ -0,0 +1,144 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include "exception_handler.hpp"
+
+// Buffer locations for MM Host interfaces
+constexpr int kBL0 = 0;
+constexpr int kBL1 = 1;
+constexpr int kBL2 = 2;
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class SimpleVAdd;
+
+struct SimpleVAddKernel {
+  sycl::ext::oneapi::experimental::annotated_arg<
+      int *, decltype(sycl::ext::oneapi::experimental::properties{
+                 sycl::ext::intel::experimental::buffer_location<kBL0>,
+                 sycl::ext::intel::experimental::dwidth<32>,
+                 sycl::ext::intel::experimental::latency<0>,
+                 sycl::ext::intel::experimental::read_write_mode_read,
+                 sycl::ext::intel::experimental::wait_request_requested})>
+      A_in;
```

Comment:
snake_case here and below
```suggestion
      a_in;
```